### PR TITLE
Fixed issue #18423: SQL error when quota enabled on postgresql

### PR DESCRIPTION
**Pull request by Shnoulle**
Dev: check if <>0

### CODE REVIEW

**Review by gabrieljenik — APPROVED** [review]
Code looks OK. Haven't tested it.

**Comment by Shnoulle** [comment]
Tested : https://bugs.limesurvey.org/view.php?id=18423#c73640

**Comment by olleharstedt** [comment]
It looks a bit weird, doesn't it? Does it still work for MySQL?

**Comment by Shnoulle** [comment]
Else can move to compare https://github.com/yiisoft/yii/blob/9693985d4e826b49bdb9f34c79ea5ae2389349e8/framework/db/schema/CDbCriteria.php#L428 …

File: application/helpers/frontend_helper.php [hunk 1]
```diff
@@ -1865,7 +1865,7 @@ function checkCompletedQuota($surveyid, $return = false)
             // Answers are the same in quota + an answer is submitted at this time (bPostedField)
             //  OR all questions is hidden (bAllHidden)
             $bAllHidden = QuestionAttribute::model()
-                ->countByAttributes(array('qid' => $aQuotaQid), 'attribute=:attribute AND value=:value', array(':attribute' => 'hidden',':value' => 1)) == count($aQuotaQid);
+                ->countByAttributes(array('qid' => $aQuotaQid), 'attribute=:attribute AND value=:value', array(':attribute' => 'hidden',':value' => '<>0')) == count($aQuotaQid);
```
Comment (olleharstedt) [inline application/helpers/frontend_helper.php:1868]: value = :value 

but

:value = <>0

so

value = <> 0 ?

**Comment by Shnoulle** [comment]
I don't hav time currently to ad a Test for hidden quota and 0 quota … must do it.

**Comment by Shnoulle** [comment]
https://limesurvey-4.limesurvey.org/view.php?id=18423#c74415

> https://limesurvey-4.limesurvey.org/view.php?id=18423#c74415

**Comment by olleharstedt** [comment]
You want this fix in LS5 too? Master is LS6.

**Comment by Shnoulle** [comment]
cherry-pick :)

**Pull request merged**